clang/lib/Driver/ToolChains/Cuda.cpp

```diff
@@ -1151,7 +1151,9 @@ CudaToolChain::TranslateArgs(const llvm::opt::DerivedArgList &Args,
   }
 
   for (Arg *A : Args) {
-    DAL->append(A);
```

Comment:
nit: Adding a comment here could be helpful.
Please add a test to check if `-Xcuda-ptxas` is recognized in `clang-cl` mode as well as a test to check if the arguments are passed to the `PTXAS` just once.